javascript/HtmlEditorField.js

```diff
@@ -1158,8 +1158,17 @@ ss.editorWrappers['default'] = ss.editorWrappers.tinyMCE;
 				var urlField = this.getURLField(), container = this.closest('.CompositeField'), form = this.closest('form');
 
 				if (urlField.validate()) {
+
+					var val = urlField.val(), orig = val;
+
 					container.addClass('loading');
-					form.showFileView('http://' + urlField.val()).done(function() {
+
+					// add "http://" if a protocol is missing from the url
+					if (val.match(/^https?:\/\//i) == void 0) {
```

Comment:
Allowing scheme-less protocols for this feature could be a security risk; I'd rather mandate that it was included.

For instance, there are cases wher $_SERVER['REQUEST_URI'] could be populated with a relative url that looks like a scheme-less absolute url.

http://www.mysite.com//page would have a REQUEST_URI val of //page, which when placed directly in a link would act as an absolute url (not a relative url).

In any case, although I'm not sure where this could surface exactly, it's another risk you need to investigate if you allow users to omit protocol. :P

It could also mess with fixes to the above security issue, which might in some cases mistakenly evaluate those urls to relative ones.